### ISSUE TITLE
Cache translations from GlotPress for 24 hours in CI instead of pulling on every single CI run

### ISSUE THREAD

**Issue by sarayourfriend** [WordPress/openverse#4614]
## Current Situation

It's possible for our CI/CD suite to run enough times for us to get rate limited when downloading translations from GlotPress. See this run, for example:

https://github.com/WordPress/openverse/actions/runs/9937426887/job/27447855989?pr=4585#step:4:1007

We pull translations from GlotPress every time we build the frontend image, which happens any time frontend changes occur. We also pull it whenever storybook is built (because storybook build runs the `build` script, and `prebuild` runs `i18n` which fetches translations). Storybook is built for documentation site changes as well as relevant frontend changes.

For certain change sets, we could conceivably fetch translations from GlotPress four times:
- First to build the frontend image
- Next to check storybook build
- Next to run storybook playwright tests
- Next to build the documentation site

## Suggested Improvement

The easiest change to make would be to remove translation getting from storybook. We do not currently have any means of changing the language used to render components in storybook, as far as I can tell. The RTL/LTR toggle just changes the direction, but leaves the English language text—[see this story, for example, which uses text from the translation files rather than hard-coded text from Storybook variables](https://docs.openverse.org/storybook/?path=/story/components-vcontentreportpopover--default&globals=languageDirection:rtl); there is no way to change the language! **We could go ahead and do that as the easy step-one to prevent the most common case of unnecessary translation pull.**

That would still result in requesting the translations every time the frontend image is built, which could be very often if a PR is pushed many times, or if a lot of people are working at one time.

Realistically, there's no reason to pull translations so often. If we pulled translations in CI once every 24 hours and cached it in a shared artefact instead, it would speed up CI (even if only marginally), and prevent this problem from happening altogether, regardless of how active our repository is. A 24 hours SLA of translation changes is not unreasonable, and if there was a problem where we needed to pull translations right away and push a new build, we could easily manually dispatch the translation pull workflow and build a new frontend image.

## Benefit

Prevent rate limiting to GlotPress and slightly speed up frontend (and storybook, so therefore also documentation site) build times in CI.

**Comment by obulat** [comment]
We have several stories with `-rtl` snapshots, but some only have it in their name, not even changing the HTML direction (`VCollectionHeader`). We do have stories with full RTL support (we need to add the language switcher to the story to do that): https://docs.openverse.org/storybook/?path=/story/components-vmediainfo-vmediareuse--default&globals=languageDirection:rtl

As for the issue of getting translations, I think we don't download the translations from GlotPress for Playwright tests. To run the tests we use the `prod:playwright` command that copies the test locales and then runs `build:only`.

https://github.com/WordPress/openverse/blob/90f87b8712b50cf05401de91eb5805b067efeb85/frontend/package.json#L21

**Comment by sarayourfriend** [comment]
@obulat I'm talking about the storybook tests, not the playwright tests. The playwright tests use the test locales and do not require the live translations at all.

**Comment by obulat** [comment]
I was also thinking about storybook tests, but didn't think about the different commands that are used in Playwright and Storybook (Playwright) tests.

Storybook tests re-use the Playwright configs, but they run the `storybook build` command, which under the hood uses the `pnpm build` to build the app. I don't know if it's possible to change what `storybook build` does (whether it uses the `build` or `build:only` command). Maybe we could set an env variable when running the Storybook Playwright tests, and then, the i18n scripts would check that variable and skip download if it's set?

**Comment by sarayourfriend** [comment]
Right, that would address the stop gap I described in the issue, just reducing the number of places we query for translations.

I don't think that's a replacement for caching it altogether though. It's still conceivable that we would have many builds happening close together that we make a bunch of requests to GlotPress and get rate limited.

**Comment by dhruvkb** [comment]
Another way that doesn't involve caching is to download the translations in a job (that only runs if the frontend has any changes), and then export them as an artifact that other jobs can use. That way one CI run only downloads translations once, at max, and zero times if the frontend has not changed.

We can also request GlotPress to not increase the limits or not rate limit calls from GitHub Actions worker IPs but that would be a bit time-consuming and difficult.

**Comment by sarayourfriend** [comment]
> Another way that doesn't involve caching is to download the translations in a job (that only runs if the frontend has any changes), and then export them as an artifact that other jobs can use. That way one CI run only downloads translations once, at max, and zero times if the frontend has not changed.

That's not really necessary or an overall improvement if taken in consideration with the changes to remove downloading translations from storybook, it's the only place other than the frontend build that pulls from GlotPress.

As mentioned in the issue body, removing them from storybook is at least cutting the total possible number to 1 rather than 4 for a single CI run.

Is caching the translations a bad idea, something we shouldn't do, for some reason? We could stop pulling translations for builds not on `main` (branch and PR builds rather than merge builds), if we absolutely need published builds to always pull the latest translations. That would remove any complexity here. It's not like the POT/JSON files are a core part of testing the build, and we can still pull them if the changeset includes changes to the translation _scripts_, so we can verify their functionality.

If _caching_ is specifically something @dhruvkb and @obulat y'all don't think is a good idea, please say so directly and we can close it in favour of https://github.com/WordPress/openverse/issues/4621.

**Comment by dhruvkb** [comment]
I haven't actually given adequate serious thought to know if caching would cause any problems with VRT or deployments. I'll do that before commenting on that topic.

I suggested the artifact approach because it will definitely reduce the number of GlotPress calls and will definitely not cause problems so it is a good, but not perfect, step towards our goal.
